### PR TITLE
Fixes xml generation

### PR DESCRIPTION
**Pull request by philnash**
This leads to XML that has line breaks after each element, which wasn't present before, but doesn't break anything in terms of the expected output. This follows #382 (well, it includes all of #382, but should be separately reviewed), which updated the tests to not worry about insignificant whitespace. It should also be written into Yoyodyne since `twiml.rb` is generated.

The purpose of this PR is to properly use all of Nokogiri's features to generate the XML, including the header, and fixes what seems like an inconsistency, in which the XML is generated from a `Node` instead of the whole `Document`.

### CODE REVIEW

File: spec/twiml/messaging_response_spec.rb [hunk 1]
```diff
@@ -3,93 +3,185 @@
 describe Twilio::TwiML::MessagingResponse do
   context 'Testing Response' do
     it 'should allow empty response' do
-      r = Twilio::TwiML::MessagingResponse.new
-      expect(r.to_s).to eq('<?xml version="1.0" encoding="UTF-8"?><Response/>')
+      expected_doc = parse <<-XML
+        <Response/>
+      XML
```
Comment (efossier) [inline spec/twiml/messaging_response_spec.rb]: 😮

File: spec/twiml/messaging_response_spec.rb [hunk 2]
```diff
@@ -3,93 +3,185 @@
...
+      expected_doc = parse <<-XML
+        <Response>
+          <Message>Hello</Message>
+          <Redirect>example.com</Redirect>
+        </Response>
+      XML
+      response = Twilio::TwiML::MessagingResponse.new
+      response.message(body: 'Hello')
+      response.redirect('example.com')
+      doc = parse(response)
```
Comment (efossier) [inline spec/twiml/messaging_response_spec.rb]: do we not need to call `to_s` on the response first?

Comment (philnash) [reply spec/twiml/messaging_response_spec.rb]: The [`parse` method](https://github.com/twilio/twilio-ruby/pull/386/files#diff-93830fa29d616f7c87903d08b5b1b29aR36) calls `to_s` to save on repetition.

Comment (efossier) [reply spec/twiml/messaging_response_spec.rb]: gotcha

**Pull request merged**